q7/main.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+/* global inputWithCheck getRandomInt output */
+
+const MATERIAL_NUMBERS = ['0', '1', '2', '3', '4', '5', '6', '7', '9']
+const LENGTH = 4
+const MESSAGE = '4桁の数字は？'
+main()
+function main() {
+  const base = getRandomNumbers(LENGTH)
+  try {
+    const count = hitAndBlow(base)
+    output(count + ' 回で正解!')
+  } catch (e) {
+    output(e.message)
+    output('正解は' + base.toString() + 'でした')
+  }
+}
+/**
+ * @param {String} base 基準
+ * @param {Number} count 試行回数
+ */
+function hitAndBlow(base, count = 0) {
+  const input = inputWithCheck(MESSAGE, function(input) {
+    return !isNaN(input) && LENGTH === input.length
+  })
+  output(input)
+  const result = checkHitAndBlow(base, input)
+  count++
+  if (LENGTH !== result.hit) {
+    output('外れ：' + result.hit + ' Hit, ' + result.blow + ' Blow')
+    return hitAndBlow(base, count)
+  }
+  return count
+}
+/**
+ * 指定した桁数の数値を格納した配列（各桁の数値は重複しない）
+ * @param {Number} len 指定桁数
+ * @param {String} str 配列
+ */
+function getRandomNumbers(len, str = '') {
+  if (len === str.length) return str
+  const elementNumber = getRandomInt(9 - str.length)
```

Comment:
せっかく汎用的なメソッドにしているので、マジックナンバーを排除したほうがより良いと思います。
```suggestion
  const elementNumber = getRandomInt(MATERIAL_NUMBERS.length)
```

あと、ここも0-9の数値が欲しいので、引数は10からの開始になりますね。

Reply:
なるほど。
spliceしてるので、その配列の長さがそのまま使えるのか。

71fb773